source/patterns/@aws-solutions-constructs/aws-lambda-kendra/lib/index.ts

```diff
@@ -0,0 +1,197 @@
+   * @since 1.120.0
+   * @access public
+   */
+  constructor(scope: Construct, id: string, props: LambdaToKendraProps) {
+    super(scope, id);
+    defaults.CheckProps(props);
+
+    if (props.deployVpc || props.existingVpc) {
+
+      this.vpc = defaults.buildVpc(scope, {
+        defaultVpcProps: defaults.DefaultIsolatedVpcProps(),
+        existingVpc: props.existingVpc,
+        userVpcProps: props.vpcProps,
+        constructVpcProps: {
+          enableDnsHostnames: true,
+          enableDnsSupport: true,
+        },
+      });
+
+      defaults.AddAwsServiceEndpoint(scope, this.vpc, defaults.ServiceEndpointTypes.KENDRA);
+    }
+
+    // Setup the Lambda function
+    this.lambdaFunction = defaults.buildLambdaFunction(this, {
+      existingLambdaObj: props.existingLambdaObj,
+      lambdaFunctionProps: props.lambdaFunctionProps,
+      vpc: this.vpc,
+    });
+
+    this.kendraIndex = defaults.buildKendraIndex(this, id, {
+      kendraIndexProps: props.kendraIndexProps,
+      existingIndexObj: props.existingKendraIndexObj
+    });
+
+    this.kendraDataSources = defaults.AddMultipleKendraDataSources(this, id, this.kendraIndex, props.kendraDataSourcesProps);
+
+    // Update Lambda function IAM policy with correct privileges to Kendra index
+    const normalizedPermissions = props.indexPermissions ? defaults.normalizeKendraPermissions(props.indexPermissions) : undefined;
+    if (!normalizedPermissions || normalizedPermissions.includes("READ")) {
+      // Add policy with query permissions
+      this.lambdaFunction.addToRolePolicy(
+        new iam.PolicyStatement({
+          actions: [
+            "kendra:Query",
+            "kendra:Describe*",
+            "kendra:Get*",
+            "kendra:BatchGet*",
+            "kendra:List*",
+            "kendra:Retrieve"],
+          resources: [this.kendraIndex.attrArn]
+        })
+      );
+    }
+
+    if (!normalizedPermissions || normalizedPermissions.includes("SUBMITFEEDBACK")) {
+      // Add policy with query permissions
+      this.lambdaFunction.addToRolePolicy(
+        new iam.PolicyStatement({
+          actions: [
+            "kendra:SubmitFeedback"],
+          resources: [this.kendraIndex.attrArn]
+        })
+      );
+    }
+
+    if (normalizedPermissions?.includes("WRITE")) {
+      // Add policy with query permissions
+      this.lambdaFunction.addToRolePolicy(
+        new iam.PolicyStatement({
+          actions: [
+            "kendra:Associate*",
+            "kendra:BatchPut*",
+            "kendra:Clear",
+            "kendra:Create*",
+            "kendra:Delete*",
+            "kendra:Disassociate*",
+            "kendra:Put*",
+            "kendra:Update*",
+            "kendra:Start*",
+            "kendra:Submit*",
+            "kendra:Stop*",
+            "kendra:TagResource",
+            "kendra:UntagResource"
+          ],
+          resources: [this.kendraIndex.attrArn]
+        })
+      );
+    }
```

Comment:
These permissions are all specific to Kendra? Do we foresee having another kendra-based construct in the future? If so, consider moving this block into the kendra-helper. I guess we could always do it `if/when` we have a second kendra-based construct.